File: CS/XPServerModeViewDemo/ProgressForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace XPServerModeViewDemo {
    public partial class ProgressForm : DevExpress.XtraEditors.XtraForm, IProgress<int> {
        public ProgressForm() {
            InitializeComponent();
        }

        public void Report(int value) {
            progressBarControl1.EditValue = value;
        }
    }
}

File: VB/XPServerModeViewDemo/ProgressForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Linq
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace XPServerModeViewDemo
	Partial Public Class ProgressForm
		Inherits DevExpress.XtraEditors.XtraForm
		Implements IProgress(Of Integer)

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub Report(ByVal value As Integer) Implements IProgress(Of Integer).Report
			progressBarControl1.EditValue = value
		End Sub
	End Class
End Namespace